src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -251,145 +282,160 @@ public TurretSubsystem(TankSubsystem tankSubsystem, JetsonConnection connection)
+                .addListener("Manual turn ref", turntableRefPos, this::setTurntableRefPos, 6, 3)
+                .addListener("Manual hood ref", hoodRefPos, this::setHoodRefPos, 7, 3);
         }
     }
 
     @Override
     public void periodic() {
-        shuffleboardTurntablePosEntry.setValue(Math.toDegrees(turntableEncoder.getPosition()));
-        shuffleboardFlywheelVeloEntry.setValue(flywheelEncoder.getVelocity());
-        shuffleboardHoodPosEntry.setValue(Math.toDegrees(hood.getSelectedSensorPosition() / HOOD_RADIANS_TO_TICKS));
+        turntablePosEntry.setValue(Math.toDegrees(turntableEncoder.getPosition()));
+        flywheelVeloEntry.setValue(flywheelEncoder.getVelocity());
+        hoodPosEntry.setValue(Math.toDegrees(hood.getSelectedSensorPosition() / HOOD_RADIANS_TO_TICKS));
+        jetsonDetectedEntry.setValue(jetson.turretVisionWorking());
+        driverOverrideEntry.setValue(driverOverrideFlywheel);
+        turretModeEntry.setValue(mode.toString());
 
         // Check limit switches and reset encoders if detected
         // if (leftLimitSwitch.get()) turntableEncoder.setPosition(TURNTABLE_MIN_RADIANS);
         // if (rightLimitSwitch.get()) turntableEncoder.setPosition(TURNTABLE_MAX_RADIANS);
 
         Pose2d currentPosition = tankSubsystem.getRobotPosition();
         boolean runFlywheel = (/* !tankSubsystem.isMoving() && */ ballReady) || driverOverrideFlywheel;
+        runFlywheelEntry.setValue(runFlywheel);
 
         // Set turntable lazy tracking if a ball isn't ready
         double pow = !ballReady ? 0.25 : 0.5;
         turntablePidController.setOutputRange(-pow, pow);
 
-        // If the hub is in vision range, use vision's `r` and `theta` as ground truth.
-        // While the flywheel is running, use the manual system values instead to
-        // prevent camera issues while the flywheel shakes the turntable.
+        Pair<Double, Double> deltas = calculateRThetaDeltas(previousPosition, currentPosition);
+
+        // If the hub is in vision range and the jetson has fresh data, use vision's `r` and `theta` as ground truth.
+        // While the flywheel is running, use the manual system values instead to prevent camera issues while the 
+        // flywheel shakes the turntable.
         // If the jetson has been disabled on shuffleboard, use `rtheta` instead.
-        if (jetson.turretVisionWorking() && !runFlywheel && !jetsonDisabled) {
-            r = jetson.getHubDistance();
-            theta = jetson.getTurretTheta();
+        if (jetson.turretVisionWorking() && (!runFlywheel || mode == TurretMode.RETRACTED) && !jetson.getConsumed() && !jetsonDisabled) {
+            Pair<Double, Double> data = jetson.getData();
+            r = data.getFirst();
+            theta = Math.PI + data.getSecond() - turntableEncoder.getPosition();
 
-            // System.out.println("JETSON r: " + r + " theta: " + theta);
+            // Reset offsets when we refresh rtheta from vision.
+            resetOffsets();
```

Comment:
idt we should reset offsets, we should try not to and see how driver practice goes. Because ideally offset accounts for bad interpolation tuning, which will persist even if new vision data comes in

Reply:
Hmm, I thought offsets were for rtheta drifting which would be reset when vision sends data?